test/geometry/test_quaternion.py

```diff
@@ -221,3 +222,134 @@ def test_slerp(self, device, dtype, batch_size):
                 q3 = q1.slerp(q2, t)
                 q4 = Quaternion.from_axis_angle(axis * t * 3.14159)
                 self.assert_close(q3, q4)
+
+
+class TestToEuler(BaseTester):
+    def test_smoke(self, device, dtype):
+        q = Quaternion.random(batch_size=1)
+        q = q.to(device, dtype)
+        roll, pitch, yaw = q.to_euler(q.w, q.x, q.y, q.z)
```

Comment:
this doesn't seem correct according the api

Reply:
I believe everything should be there now.